Assets/Scripts/RoadControl.cs

```diff
@@ -0,0 +1,616 @@
+    private int curRoadStripSpriteIndex;
+    private Sprite curRoadStripSprite;
+    private float distCamToScreen;
+    private float nearEdgeWidthScale, farEdgeWidthScale;
+
+    private List<RoadSegment> roadSegments = new List<RoadSegment>();
+    
+    [SerializeField]
+    public List<RoadCurve> roadCurves;
+    [SerializeField]
+    public RoadCurve rdCurveExample;
+
+    private int curSegmentIndex;
+
+    [SerializeField]
+    private float topSpeed, autopilotSpeed, acceleration, deceleration, maxTurning, maxArmor;
+    private float curPlayerSpeed, curPlayerTurning, camCurZ, playerZoffset, curPlayerPosX, curArmor;
+
+    // GAME STATES
+    private enum GameState { InMenu, TrackStart, Driving, Crashing, Exploding, TrackComplete, GameOver }
+    private GameState GameMode;
+
+    private bool canDrive, isOnAutopilot, isSlipping, isOffRoading, isCrashing, isSpinningOut, isExploding, isRolling;
+
+    // ENEMIES
+
+    // BILLBOARDS and OTHER SPRITES
+    public List<BillboardSprite> billboardData;
+
+    private List<BillboardSprite> billboardDataExpanded = new List<BillboardSprite>();
+    private List<BillboardSprite> visibleBillboards = new List<BillboardSprite>();
+
+    [SerializeField]
+    private float billboardSizeCorrectionFactor;
+
+    // Menu
+    [SerializeField]
+    private RawImage menuBgd;
+
+    // HUD
+    [SerializeField]
+    private RawImage speedometer, speedometerNeedle, progressTrack, progressCar, armor, armorFrame;
+    [SerializeField]
+    private Text speedometerReading, timer, timerLabel, armorLabel, messageText;
+    [SerializeField]
+    private string trackStartReadyMessage, trackStartGoMessage, gameOverMessage, trackCompleteMessage; 
+    [SerializeField]
+    private float speedometerRotationFactor, timeLimit;
+    private float timeElapsed, lastStartTime;
+    [SerializeField]
+    private float messageDelay;
+    private float messageDelayRefTime;
+    private bool readyMessageShown, goMessageShown, victoryMessageShown, gameOverMessageShown;
+    private float progressCarInitialY;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        playerCarSprite = playerCar.GetComponent<SpriteRenderer>();
+
+        if (billboardData.Count > 0) {
+            foreach (BillboardSprite bb in billboardData) {
+                if (bb.howMany > 1) {
+                    for (int i = 0; i < bb.howMany; i++) {
+                        BillboardSprite bbTemp = bb;
+                        bbTemp.howMany = 1;
+                        bbTemp.segmentIndex = bb.segmentIndex + 1;
+                        billboardDataExpanded.Add(bbTemp);
+                    }
+                }
+                else {
+                    billboardDataExpanded.Add(bb);
+                }
+            }
+        }
+
+        curPlayerSpeed = 0;
+        camCurZ = 0;
+        distCamToScreen = (numScreenLines / 2f) / (Mathf.Tan(FOV / 2f));
```

Comment:
Cool trick with `dividing by 2`

If you are using integers, try a `right bit shift` (`value >> 1`) as it is many times faster than dividing (binary math can be pretty handy sometimes). You can also use `left bit shift` (`<< 1`) to rapidly multiply integers by 2.

If you are using floats, try multiplying by 0.5f instead as that is also a little faster than division.

But to be honest, most good compilers will change the divide by 2 lines to something like I mentioned or use other tricks to produce faster code.